stdlib/public/core/FloatingPoint.swift

```diff
@@ -1908,6 +1908,62 @@ extension BinaryFloatingPoint {
       self = Self(value_)
 #endif
     default:
+      if value.isFinite {
+        // According to IEEE 754:
+        // - The set of finite floating-point numbers representable within a
+        //   particular format is determined by: the radix (b), the precision
+        //   (p, the number of digits in the significand), and the maximum and
+        //   minimum exponent (emax and emin, respectively, where
+        //   emin = 1 - emax).
+        // - In a binary interchange format, each floating-point number has only
+        //   one encoding.
+        //
+        // If two binary interchange formats have the same exponent bit count
+        // (w) and significand bit count (p - 1), then they must share the same
+        // encoding for finite values.
+        switch (Source.exponentBitCount, Source.significandBitCount) {
+#if !os(macOS) && !(os(iOS) && targetEnvironment(macCatalyst))
+        case (5, 10):
+          if #available(iOS 14.0, watchOS 7.0, tvOS 14.0, *) {
+            let value_ = Float16(
+              sign: value.sign,
+              exponentBitPattern: Float16.RawExponent(value.exponentBitPattern),
```

Comment:
All of these should probably use `truncatingInNeeded:` rather than the unlabelled init.

Reply:
Yup, I waffled between actively embracing the implied assertion (which should never be tripped) or using `truncatingIfNeeded`; I'll take the latter approach.